pkg/code-generator/codegen/templates/resource_template.go

```diff
@@ -38,12 +38,17 @@ var (
 {{- if $.IsCustom }}
 	_ resources.CustomInputResource = new({{ .Name }})
```

Comment:
CustomInputResources extend Inputresources (things with statuses) by allowing the definition of custom marshaling behavior. if we change this template to create a hashableInputResource, i think we'll need to update the CustomInputResource as well

Reply:
I think it is reasonable to get rid of the hashable input resource, or we could make `InputResource` interface with `HashableResource` instead of `Resource`.